### PR TITLE
Initial work of GraphQL Engine Rewrite

### PR DESCRIPTION
**Pull request by bryan-coursera**
cc @yifan-coursera and @vishalkuo

### CODE REVIEW

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/filters/DefaultFilters.scala [hunk 1]
```diff
@@ -8,5 +8,5 @@ class DefaultFilters @Inject() (
     queryComplexityFilter: QueryComplexityFilter,
     engineMetricsFilter: EngineMetricsFilter)
   extends FilterList(List(
-    queryComplexityFilter,
+//    queryComplexityFilter,
```
Comment (vishalkuo) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/filters/DefaultFilters.scala]: Will we remove this entirely? Or is this a temp comment

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeResourceUtils.scala [hunk 2]
```diff
@@ -116,4 +128,60 @@ object NaptimeResourceUtils extends StrictLogging {
   def formatResourceName(resourceName: ResourceName): String = {
     s"${resourceName.topLevelName.capitalize}V${resourceName.version}"
   }
+
+
+  /**
+    * This regex is used to match both "$instructorIds" and "${instructorDetails/instructorIds}"
+    */
+  val InterpolationRegex =
```
Comment (vishalkuo) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/schema/NaptimeResourceUtils.scala]: Can this be made private?

File: naptime/src/main/scala/org/coursera/naptime/ari/engine/EngineHelpers.scala [hunk 3]
```diff
@@ -109,6 +112,31 @@ object EngineHelpers extends StrictLogging {
     }
   }
 
+  // TODO(bryan): Dedupe this from NaptimeUnionField
```
Comment (vishalkuo) [inline naptime/src/main/scala/org/coursera/naptime/ari/engine/EngineHelpers.scala]: With the removal of `EngineImpl`, should this file (and possibly this package) be renamed? `utilities` or something similar?

File: naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala [hunk 4]
```diff
@@ -112,53 +112,29 @@ class GraphQLController @Inject() (
...
-                  }
-              }.recover {
-                case error: QueryAnalysisError =>
-                  OutgoingQuery(error.resolveError.as[JsObject], None)
-                case error: ErrorWithResolver =>
-                  OutgoingQuery(error.resolveError.as[JsObject], None)
-                case error: Exception =>
-                  logger.error("GraphQL execution error", error)
-                  OutgoingQuery(Json.obj("errors" -> Json.arr(error.getMessage)), None)
+            val context = SangriaGraphQlContext(Response.empty, fetcher, requestHeader, ec)
```
Comment (vishalkuo) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala]: Do we plan on adding a `metricsCollector` back in at some point?

Comment (bryan-coursera) [inline naptime-graphql/src/main/scala/org/coursera/naptime/ari/graphql/controllers/GraphQLController.scala]: Eventually, although the actual metrics that are collected will need to be very different now.

**Pull request merged**